lib/nats.js

```diff
@@ -801,9 +817,12 @@ Client.prototype.flush = function(opt_callback) {
 
 Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
   if (!msg) { msg = EMPTY; }
+  if (typeof subject !== 'string') {
+    this.emit('error', PUB_BAD_SUBJECT_ERR);
+  }
```

Comment:
In Node, I take that as a presumption that the calling party doesn't care about the transmission, so nothing happens.

Actual errors on the connection (which include loss and failures) are handled by the stream/connection/client eventemitter.  Those are events unrelated to requests.  Most of that is already handled for you by the stream and connection.